gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -487,8 +487,10 @@ private static void cleanupStagingDataPerTask(JobState jobState) {
     Map<String, ParallelRunner> parallelRunners = Maps.newHashMap();
     try {
       for (TaskState taskState : jobState.getTaskStates()) {
+        TaskState fatTaskState = new TaskState(taskState);
+        fatTaskState.addAllIfNotExist(jobState);
```

Comment:
It seems it's pretty tricky to add this into `ParallelRunner`.  I think this should be fine because this is entirely in memory, so it should be much less significant than deleting files from HDFS.